pkg/ext-proc/backend/pod_reconciler.go

```diff
@@ -56,15 +55,17 @@ func (c *PodReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(c)
 }
 
-func (c *PodReconciler) updateDatastore(k8sPod *corev1.Pod, inferencePool *v1alpha1.InferencePool) {
-	pod := Pod{
-		Name:    k8sPod.Name,
-		Address: k8sPod.Status.PodIP + ":" + strconv.Itoa(int(inferencePool.Spec.TargetPortNumber)),
-	}
-	if !k8sPod.DeletionTimestamp.IsZero() || !c.Datastore.LabelsMatch(k8sPod.ObjectMeta.Labels) || !podIsReady(k8sPod) {
-		c.Datastore.pods.Delete(pod)
+func (c *PodReconciler) updateDatastore(logger logr.Logger, pod *corev1.Pod) {
+	namespacedName := types.NamespacedName{Name: pod.Name, Namespace: pod.Namespace}
+	if !pod.DeletionTimestamp.IsZero() || !c.Datastore.PoolLabelsMatch(pod.Labels) || !podIsReady(pod) {
+		logger.V(logutil.DEFAULT).Info("Pod removed or not added", "name", namespacedName)
+		c.Datastore.PodDelete(namespacedName)
 	} else {
-		c.Datastore.pods.Store(pod, true)
```

Comment:
This is another bug where we could be overriding the state of an existing metrics state.